mllib/src/main/scala/org/apache/spark/mllib/linalg/distributed/RowMatrix.scala

```diff
@@ -538,20 +538,29 @@ class RowMatrix @Since("1.0.0") (
     val col = numCols().toInt
     // split rows horizontally into smaller matrices, and compute QR for each of them
     val blockQRs = rows.glom().map { partRows =>
-      val bdm = BDM.zeros[Double](partRows.length, col)
-      var i = 0
-      partRows.foreach { row =>
-        bdm(i, ::) := row.asBreeze.t
-        i += 1
+      if (partRows.length == 0) {
```

Comment:
Can you just flatMap this? so that you don't have to handle None below? or filter then map?